### ISSUE TITLE
Pausing on introducer removes device on other devices

### ISSUE THREAD

**Issue by calmh** [syncthing/syncthing#4897]
Didn't see this myself, but sounds serious enough to investigate:

```
23:39:57 <_aD> hmm, if an introducer pauses all folders that it shares with a device X,
               other devices that have been introduced remove that device X
```

**Comment by imsodin** [comment]
I do have an idea how it happens and wanting to write a test as reproducer I potentially found a flaw in the current introducer tests that renders them useless (i.e. tests fail when corrected): #4898

**Comment by ProactiveServices** [comment]
You beat me to reporting it :-)
Thankfully, once folders were resumed, devices/shares were restored.
